### PR TITLE
mds/MDBalancer: cleanup

### PR DESCRIPTION
**Pull request by renhwztetecs**
Signed-off-by: huanwen ren <ren.huanwen@zte.com.cn>

### CODE REVIEW

File: src/common/config_opts.h [hunk 1]
```diff
@@ -475,7 +475,7 @@ OPTION(mds_log_events_per_segment, OPT_INT, 1024)
 OPTION(mds_log_segment_size, OPT_INT, 0)  // segment size for mds log, default to default file_layout_t
 OPTION(mds_log_max_segments, OPT_U32, 30)
 OPTION(mds_log_max_expiring, OPT_INT, 20)
-OPTION(mds_bal_sample_interval, OPT_FLOAT, 3.0)  // every 5 seconds
+OPTION(mds_bal_sample_interval, OPT_DOUBLE, 3.0)  // every 5 seconds
```
Comment (xiexingguo) [inline src/common/config_opts.h]: Can you fix the comments also?

Comment (renhwztetecs) [inline src/common/config_opts.h]: yes, thanks!

File: src/mds/MDBalancer.cc [hunk 2]
```diff
@@ -162,7 +162,10 @@ mds_load_t MDBalancer::get_load(utime_t now)
   ifstream cpu("/proc/loadavg");
   if (cpu.is_open())
     cpu >> load.cpu_load_avg;
-
+  else
+    derr << "input file '/proc/loadavg' not found" << dendl;
+  cpu.close();
```
Comment (jcsp) [inline src/mds/MDBalancer.cc]: The close() would go inside the is_open() conditional, calling close on a non-open ifstream is an error.

File: src/mds/MDBalancer.cc [hunk 3]
```diff
@@ -337,6 +340,7 @@ void MDBalancer::queue_split(CDir *dir)
 
 void MDBalancer::queue_merge(CDir *dir)
 {
+  assert(mds->mdsmap->allows_dirfrags());
```
Comment (jcsp) [inline src/mds/MDBalancer.cc]: This is incorrect, allows_dirfrags controls whether splitting is permitted.  If it is switched off again after some splits have happened, merges are allowed even though allow_dirfrags is false

File: src/mds/MDBalancer.cc (hunk 2 above)
Comment (renhwztetecs) [inline src/mds/MDBalancer.cc]: I change it :)
the destructor will release it, do not need to display the call “cpu.close()”

File: src/mds/MDBalancer.cc (hunk 3 above)
Comment (renhwztetecs) [inline src/mds/MDBalancer.cc]: this scene I did not take into account,
thanks for you reviews

**Comment by renhwztetecs** [comment]
done :)

**Pull request merged**